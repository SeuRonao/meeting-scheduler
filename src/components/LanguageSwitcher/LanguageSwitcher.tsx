import { Stack } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import i18n from "../../i18n/i18n";
import languages from "../../i18n/languages/languages.json";

export default function LanguageSwitcher() {
  const { t } = useTranslation("translation", {
    keyPrefix: "components.LanguageSwitcher",
  });
  return (
    <Stack className="justify-content-center" direction="horizontal" gap={2}>
      <div>{t("available-in")}:</div>
      <div>
        {languages
          .filter((value) => {
            return i18n.language !== value.code;
          })
          .map((value, index) => {
            return (
              <a
                href="/"
                onClick={() => {
                  i18n.changeLanguage(value.code);
                }}
                key={index}
              >
                {value.name}
              </a>
            );
          })}
      </div>
    </Stack>
  );
}
