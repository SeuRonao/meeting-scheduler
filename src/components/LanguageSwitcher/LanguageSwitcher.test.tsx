import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LanguageSwitcher from "./LanguageSwitcher";

describe("Testing LanguageSwitcher", () => {
  test("renders correctly", () => {
    const component = renderer.create(
      <MemoryRouter>
        <LanguageSwitcher />
      </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
