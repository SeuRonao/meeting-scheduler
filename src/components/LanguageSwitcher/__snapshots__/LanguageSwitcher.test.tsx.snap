// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LanguageSwitcher renders correctly 1`] = `
<div
  className="justify-content-center hstack gap-2 minBreakpoint-xs"
>
  <div>
    available-in
    :
  </div>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      PortuguÃªs
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      English
    </a>
  </div>
</div>
`;
